applications/GeoMechanicsApplication/custom_utilities/solving_strategy_factory.hpp

```diff
@@ -21,91 +21,101 @@
+            const std::vector<std::string> strategy_entries = {"max_iteration",
+                                                               "compute_reactions",
+                                                               "max_line_search_iterations",
+                                                               "first_alpha_value",
+                                                               "second_alpha_value",
+                                                               "min_alpha",
+                                                               "max_alpha",
+                                                               "line_search_tolerance",
+                                                               "move_mesh_flag",
+                                                               "reform_dofs_at_each_step",
+                                                               "echo_level"};
+
+            auto strategy_parameters = ExtractStrategyParameters(rSolverSettings, strategy_entries);
+            auto result = std::make_unique<LineSearchStrategy<TSparseSpace,
+                    TDenseSpace,
+                    TLinearSolver>>(rModelPart,
+                                    scheme,
+                                    solver,
+                                    criteria,
+                                    strategy_parameters);
+            return result;
         }
 
         return nullptr;
     }
 
 private:
-    static Parameters ExtractStrategyParameters(const Parameters &rSolverSettings)
-    {
+    static Parameters ExtractStrategyParameters(const Parameters &rSolverSettings,
+                                                const std::vector<std::string> strategy_entries) {
         Parameters result;
-        std::vector<std::string> strategy_entries = {"min_iteration",
-                                                     "number_cycles",
-                                                     "increase_factor",
-                                                     "reduction_factor",
-                                                     "max_piping_iterations",
-                                                     "desired_iterations",
-                                                     "max_radius_factor",
-                                                     "min_radius_factor",
-                                                     "search_neighbours_step",
-                                                     "body_domain_sub_model_part_list",
-                                                     "loads_sub_model_part_list",
-                                                     "loads_variable_list",
-                                                     "rebuild_level"};
-        for (const std::string& entry : strategy_entries)
-        {
-            if (rSolverSettings.Has(entry))
-            {
+        for (const std::string &entry: strategy_entries) {
```

Comment:
Also here (I'm not sure how all these '&' got moved?
```suggestion
        for (const std::string& entry: strategy_entries) {
```